tasks/support/templates/appliance_installation_script.rb.erb

```diff
@@ -36,34 +39,33 @@ VMDB_DIR                 = "/var/www/miq/vmdb"
 #
 #     https://github.com/pypa/get-pip/blob/master/tasks/generate.py
 #
-MIQPERF_GEM_BUNDLE = <<-GEM
-<%= @gem_base64_string -%>
-GEM
+<% unless @solo_gem -%>
+<%= render_partial "gem_base64_string", :env_name => "MIQPERF", :gem_base64_string => @gem_base64_string -%>
+<% end -%>
 
 <% if @stackprof_gemspec and @stackprof_gem_base64_string -%>
```

Comment:
was expecting to see a `!@solo_gem` here.

maybe that is built into `@stackprof_gemspec`?

Reply:
@kbrock as it is right now, `@stackprof` and `@solo_gem` are triggered by different parts of the codebase, even though they are fundamentally the same.

For example this:

```console
$ rake extra_c_gem[stackprof] generate_install_script
```

Would generate basically the same thing as:

```console
$ rake include_stackprof generate_install_script
```

That said, consider these:

```console
# installs `rbtrace`
$ rake solo_c_gem[rbtrace] generate_install_script
```

```console
# installs `stackprof` and `rbtrace`
$ rake include_stackprof solo_c_gem[rbtrace] generate_install_script
```

```console
# installs `stackprof` twice... and probably fails
$ rake include_stackprof solo_c_gem[stackprof] generate_install_script
```

Probably poorly named, but `@solo_gem` is more meant to be "create a script that doesn't include `manageiq-performance`", and not "only install this one gem".  Naming is hard...

Reply:
Also, as mentioned under the TODO items:

> ~~* [ ] Have the `include_stackprof` task call out to a generic `extra_c_gem` task~~
> 
> Will do this last one in another pass I think.  Would rather have tests in place before doing that.

So I do have plans to make `stackprof` quack like everything else, and just have the helper task there for convenience.